### ISSUE TITLE
Implementation of the regularization loss

### ISSUE THREAD

**Issue by adosar** [fxia22/pointnet.pytorch#109]
Hello and congrats for this nice repo!

I was looking in the implementation of the regularization loss and I don't know if it matches the original paper.

In the [feature_transform_regularizer](https://github.com/fxia22/pointnet.pytorch/blob/master/pointnet/model.py#L177C1-L177C42):
```python
loss = torch.mean(torch.norm(torch.bmm(trans, trans.transpose(2,1)) - I, dim=(1,2)))
```
Why not:
```python
loss = torch.mean(torch.norm(torch.bmm(trans, trans.transpose(2,1)) - I, dim=(1,2)).pow(2))
```

In the original paper they use:
$L_\text{reg} = ||I - AA^T||^2$

**Comment by shan-zhong** [comment]
I found that the regularization loss has only been used for trans_feat, and the loss was simply added to the total loss. Why the regularization loss of trans has been ignored. And I think it may be better if the regularization loss is only used to update the parameter in T-net, rather than the whole net.
